File: LeetCode/11.py
class Solution(object):
    def maxArea(self, height):
        """
        :type height: List[int]
        :rtype: int
        """
        mx = 0
        i = 0
        j = len(height) - 1
        while i < j:
            v = min(height[i], height[j]) * (j-i)
            mx = max(mx, v)
            if height[i] < height[j]:
                i += 1
            else:
                j -= 1
        return mx


class Solution {
public:
    int maxArea(vector<int>& height) {
        int size = height.size();
        int left = 0;
        int right = size - 1;
        int ans = 0;
        while (left < right) {
            ans = max(ans, min(height[left], height[right]) * (right - left));
            if (height[left] < height[right]) {
                ++left;
            } else {
                --right;
            }
        }
        return ans;
    }
};

File: LeetCode/sword/11.cpp
class Solution {
public:
    int minArray(vector<int>& numbers) {
        int size = numbers.size();
        if(size == 0) {
            return 0;
        }else if(size == 1) {
            return numbers[0];
        }
        int left = 0, right = size - 1;
        if(numbers[left] < numbers[right]) {
            return numbers[left];
        }
        while(left < right) {
            int mid = left + ((right - left) >> 1);
            if(numbers[mid] > numbers[right]) {
                left = mid + 1;
            }else if(numbers[mid] < numbers[left]) {
                right = mid;
            }else {
                --right;
            }
        }
        return numbers[left];
    }
};